webui/react/src/pages/ExperimentList.tsx

```diff
@@ -40,11 +59,45 @@ const ExperimentList: React.FC = () => {
     { ...defaultFilters, username: (auth.user || {}).username });
   const [ filters, setFilters ] = useState<ExperimentFilters>(initFilters);
   const [ search, setSearch ] = useState('');
+  const [ selectedRowKeys, setSelectedRowKeys ] = useState<string[]>([]);
 
   const filteredExperiments = useMemo(() => {
     return filterExperiments(experiments, filters, users.data || [], search);
   }, [ experiments, filters, search, users.data ]);
 
+  const showBatch = selectedRowKeys.length !== 0;
+
+  const experimentMap = useMemo(() => {
+    return experiments.reduce((acc, task) => {
+      acc[task.id] = task;
+      return acc;
+    }, {} as Record<string, ExperimentItem>);
+  }, [ experiments ]);
+
+  const selectedExperiments = useMemo(() => {
+    return selectedRowKeys.map(key => experimentMap[key]);
+  }, [ experimentMap, selectedRowKeys ]);
+
+  const { hasActivatable, hasArchivable, hasCancelable, hasKillable, hasPausible } = useMemo(() => {
+    const tracker = {
+      hasActivatable: false,
+      hasArchivable: false,
+      hasCancelable: false,
+      hasKillable: false,
+      hasPausible: false,
+    };
+    for (let i = 0; i < selectedExperiments.length; i++) {
```

Comment:
nit: we could do an optimization here to avoid rechecking for any of the tracker items once one of that kind is already satisfied.